ament_clang_tidy/ament_clang_tidy/main.py

```diff
@@ -214,8 +214,14 @@ def start_subprocess(full_cmd):
     for filename in files:
         report[filename] = []
 
-    error_re = re.compile('(/.*?\\.(?:%s)):(\\d+):(\\d+): (?:warning:|error:)' %
-                          '|'.join(extensions))
+    # Regex explanation:
+    #   \s*\^?                   : Capture many spaces, then an optional `^`. clang_tidy output isn't great.
+    #   (\/home\/^[^:]*)         : Group capture. Everything from `/home/` up until a `:`.
+    #   (\d+)                    : Group capture. Grabs line number.
+    #   (\d+)                    : Group capture. Grabs column number.
+    #   (?:warning:|error:note:) : Non-capturing group. Matches warning, error, note.
+    #   \[(.*)\]                 : Matches and captures [<rule_name>]. Ignores any messages from clang_tidy without an ending [<rule_name>].
+    error_re = re.compile('\\s*\^?\/home\/([^:]*):(\\d+):(\\d+): (?:warning:|error:|note:).*\\[(.*)\\]')
```

Comment:
Nice inclusion of the regex segments explanation 🙇🏼

```suggestion
    # Regex explanation:
    #   \s*\^?                   : Capture many spaces, then an optional `^`. clang_tidy output isn't great.
    #   (\/home\/^[^:]*)         : Group capture. Everything from `/home/` up until a `:`.
    #   (\d+)                    : Group capture. Grabs line number.
    #   (\d+)                    : Group capture. Grabs column number.
    #   (?:warning:|error:note:) : Non-capturing group. Matches warning, error, note.
    #   \[(.*)\]                 : Matches and captures [<rule_name>]. Ignores any messages from clang_tidy without an ending [<rule_name>].
    error_re = re.compile(r'\s*\^?/home/([^:]*):(\d+):(\d+): (?:warning:|error:|note:).*\[(.*)\]')
```

I think the regex pattern is easier to read when using a raw string. I also removed a set of spurious escapes of `/` which is not a metacharacter as mentioned above.

Reply:
https://regexr.com tricked me into thinking the extra backslashes for `/home/` were necessary, but I think they can be dropped.

Reply:
Resolved in 5d5e2eda